src/components/general/DateTimeSliderValues.vue

```diff
@@ -0,0 +1,99 @@
+import * as d3 from 'd3'
+import { onMounted, ref, watchEffect } from 'vue'
+
+import { TimeSeriesData } from '@/lib/timeseries/types/SeriesData'
+
+import { ColourScale } from '@/stores/colourScales'
+
+interface Props {
+  values: TimeSeriesData[]
+  colourScale: ColourScale | null
+  height: string
+}
+const props = withDefaults(defineProps<Props>(), { height: '5px' })
+
+const container = ref<HTMLDivElement | null>(null)
+
+const viewboxWidth = 100
+const viewboxHeight = 10
+let svg: d3.Selection<SVGSVGElement, undefined, null, undefined> | null = null
+
+onMounted(() => {
+  if (!container.value) throw new Error('Container does not exist.')
+
+  // We set the viewbox width and height to arbitrary values, then stretch the
+  // SVG to the appropriate size, ignoring its aspect ratio.
+  svg = d3
+    .create('svg')
+    .attr('viewBox', `0 0 ${viewboxWidth} ${viewboxHeight}`)
+    .attr('preserveAspectRatio', 'none')
+    .style('width', '100%')
+    .style('height', '100%')
+  container.value.replaceChildren(svg.node()!)
+
+  // Only set the watcher after we have created the SVG.
+  watchEffect(updateSliderValues)
+})
+
+interface SliderEvent {
+  index: number
+  value: number | null
+}
+
+function updateSliderValues(): void {
+  if (!svg) throw new Error('SVG element has not been created.')
+
+  // We cannot create the coloured blocks without a colour scale. Once it is
+  // defined, this should execute again.
+  if (!props.colourScale) return
+
+  // Create colour scale based on the current WMS colour map.
+  const domain = props.colourScale.colourMap.map((entry) => entry.lowerValue)
+  const range = props.colourScale.colourMap.map((entry) => entry.color)
+  const scale = d3.scaleLinear<string>().domain(domain).range(range).clamp(true)
+
+  // We want the centres of the coloured blocks to align with the slider
+  // positions, so we need to start and end with half a block width.
+  const numValues = props.values.length
+  const blockWidth = viewboxWidth / (numValues - 1)
+  const widthFunc = (data: SliderEvent) => {
+    const isEdge = data.index === 0 || data.index === numValues - 1
+    return isEdge ? blockWidth / 2 : blockWidth
+  }
+  const xFunc = (data: SliderEvent) => {
+    const isStart = data.index === 0
+    return isStart ? 0 : data.index * blockWidth - blockWidth / 2
+  }
+  const colourFunc = (data: SliderEvent) => {
+    if (data.value === null) return 'rgb(0, 0, 0, 0)'
+    return scale(data.value)
+  }
+
+  // Create a <rect> element for each data point that should align nicely with
+  // the centres of the slider positions.
+  const data: SliderEvent[] = props.values.map((event, index) => ({
+    index,
+    value: event.y,
+  }))
+  svg
+    .selectAll('rect')
+    .data(data)
+    .join('rect')
+    .attr('x', xFunc)
+    .attr('width', widthFunc)
+    .attr('height', viewboxHeight)
+    .attr('fill', colourFunc)
+}
+</script>
+
+<style scoped>
+.container {
+  height: 5px;
+  display: flex;
+}
+</style>
```

Comment:
This style is currently unused as there is no class with `.container`

Reply:
Good point, I'll remove it